dissect/target/plugins/os/windows/search.py

```diff
@@ -0,0 +1,252 @@
+    def search(self) -> Iterator[SearchIndexRecords]:
+        """Yield Windows Index Search records.
+
+        Windows Search is a standard component of Windows 7 and Windows Vista, and is enabled by default. The standard (non-Windows Server)
+        configuration of Windows Search indexes the following paths: ``C:\\Users\\*`` and ``C:\\ProgramData\\Microsoft\\Windows\\Start Menu\\Programs\\*``,
+        with some exceptions for certain file extensions (see the linked resources for more information).
+
+        Parses ``Windows.edb`` EseDB and ``Windows.db`` SQlite3 databases. Currently does not parse
+        ``GatherLogs/SystemIndex/SystemIndex.*.(Crwl|gthr)`` files or ``Windows-gather.db`` and ``Windows-usn.db`` files.
+
+        The difference between the fields ``System_Date*`` and ``System_Document_Date*`` should be researched further.
+        It is unclear what the field ``InvertedOnlyMD5`` is a checksum of (record or file content?). It might be possible
+        to correlate the field ``System_FileOwner`` with a ``UserRecordDescriptor``. The field ``System_FileAttributes`` should be
+        investigated further.
+
+        No test data available for indexed Outlook emails, this plugin might not be able to handle indexed email messages.
+
+        Resources:
+            - https://learn.microsoft.com/en-us/windows/win32/search/-search-3x-wds-overview
+            - https://github.com/libyal/esedb-kb/blob/main/documentation/Windows%20Search.asciidoc
+            - https://www.aon.com/en/insights/cyber-labs/windows-search-index-the-forensic-artifact-youve-been-searching-for
+            - https://github.com/strozfriedberg/sidr
+            - https://devblogs.microsoft.com/windows-search-platform/configuration-and-settings/
+            - https://learn.microsoft.com/en-us/windows/win32/search/-search-3x-wds-included-in-index
+        """  # noqa: E501
+
+        for db_path, user_details in self.databases:
+            if db_path.suffix == ".edb":
+                yield from self.parse_esedb(db_path, user_details)
+
+            elif db_path.suffix == ".db":
+                yield from self.parse_sqlite(db_path, user_details)
+
+            else:
+                self.target.log.warning("Unknown Windows Search Index database file %r", db_path)
+
+    def parse_esedb(self, db_path: Path, user_details: UserDetails | None) -> Iterator[SearchIndexRecords]:
+        """Parse the EseDB ``SystemIndex_PropertyStore`` table."""
+
+        with db_path.open("rb") as fh:
+            db = EseDB(fh)
+            table = db.table("SystemIndex_PropertyStore")
+
+            # Translates e.g. ``System_DateModified`` to ``15F-System_DateModified``
+            COLUMNS = {col.split("-", maxsplit=1)[-1]: col for col in table.column_names}
+
+            for record in table.records():
+                values = {
+                    clean_col: record.get(full_col) for clean_col, full_col in COLUMNS.items() if record.get(full_col)
```

Comment:
This executes `get` twice. I wonder if the tradeoff from changing this to a for-loop and eliminating that second execution is faster than the bump you get from it being a comprehension.

Reply:
Also, doesn't this database have like thousands of columns? Wouldn't it be a lot faster if you create a little shim class that dynamically translates the column name lookup?

Reply:
It's probably slightly better, but I'd be curious if something like this wouldn't still be faster (pseudocode):

```python
class Shim:
    def __init__(self, table, record):
        self.table = table
        self.record = record
        self.columns = {col.split("-", maxsplit=1)[-1]: col for col in table.column_names}

    def get(self, name):
        return self.record.get(self.columns.get(name, name))
```

So translating the `.get` on the fly, instead of iterating thousands of columns only to end up reading 4 of them.